scripts-local/tests/integration_tests.bats

```diff
@@ -141,78 +155,124 @@ setup() {
-  assert_equal "$NEW_STAKE" "333"
+  diff=$(($NEW_STAKE - $OLD_STAKE))
+  assert_equal "$diff" "333"
 }
 
 # check that a second liquid staking call kicks off reinvestment
-@test "[NOT-IMPLEMENTED] rewards are being reinvested" {
-  # check the rewards balance
-  # wait a day
-  # check the withdrawal account balance
-  # wait a day
-  # check that 90% of rewards were transferred to the delegation account
-  # check that 10% of rewards were transferred to the rev EOA
-  # wait a day
-  # check that rewards were staked
+@test "[INTEGRATION-BASIC] rewards are being reinvested (delegated balance increasing)" {
+  old_staked_bal=$($GAIA_CMD q staking delegation $DELEGATION_ICA_ADDR $GAIA_DELEGATE_VAL | GETSTAKE)
+
+  # liquid stake again to kickstart the reinvestment process
+  $STRIDE_CMD tx stakeibc liquid-stake 1000 uatom --keyring-backend test --from val1 -y --chain-id $STRIDE_CHAIN
+  # wait four days (transfers, stake, move rewards, reinvest rewards)
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  BLOCK_SLEEP 2  
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+
+  # simple check that number of tokens staked increases
+  new_staked_bal=$($GAIA_CMD q staking delegation $DELEGATION_ICA_ADDR $GAIA_DELEGATE_VAL | GETSTAKE)
+  STAKED_BAL_INCREASED=$((new_staked_bal > old_staked_bal))
+  assert_equal "$STAKED_BAL_INCREASED" "1"
+}
+
+# check that exchange rate is updating
+@test "[INTEGRATION-BASIC] exchange rate is updating" {
+  # read the exchange rate
+  RR1=$($STRIDE_CMD q stakeibc list-host-zone | grep -Fiw 'RedemptionRate' | grep -Eo '[+-]?[0-9]+([.][0-9]+)?')
+
+  # wait for reinvestment to happen (3 days is enough)
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+
+  RR2=$($STRIDE_CMD q stakeibc list-host-zone | grep -Fiw 'RedemptionRate' | grep -Eo '[+-]?[0-9]+([.][0-9]+)?')
+
+  # check that the exchange rate has increased
+  MULT=1000000
+  RR_INCREASED=$(( $(FLOOR $(DECMUL $RR2 $MULT)) > $(FLOOR $(DECMUL $RR1 10000))))
+  assert_equal "$RR_INCREASED" "1"
 }
 
+
 # check that redemptions and claims work
-@test "[NOT-IMPLEMENTED] redemption works" {
+@test "[INTEGRATION-BASIC] redemption works" {
+  old_redemption_ica_bal=$($GAIA_CMD q bank balances $REDEMPTION_ICA_ADDR --denom uatom | GETBAL)
   # call redeem-stake
-  # check for an unbonding record
-  # check that a UserRedemptionRecord was created with isClaimabled = false
+  amt_to_redeem=89
```

Comment:
jc, any particular reason this is 89?

Reply:
I think this is from 3.sh, long story is we were testing at one point and started by redeeming 100 tokens, then 90 (to differentiate from the previous claim, without killing the network), then went to 89, then the test work. Short story is there's no reason it needs to be 89! :D

Reply:
Agreed, no reason to be 89 :)